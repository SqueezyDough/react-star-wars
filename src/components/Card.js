import React from 'react'
import { Utils } from '../modules/index'
import styled from 'styled-components'

function Card ({ className, character, useInitials }) {
  const name = character.name || character.title
  const characterURL = Utils.getCharacterURL(character.url)

  // take the first character or the initials
  const shortName = useInitials ? Utils.getInitials(name) : Utils.getFirstCharacter(name)

  return (
    <a href={characterURL} className={className}>
      <article className='card'>
        <span className='card__short-title'>{shortName}</span>
        <h1 className='card__title'>{name}</h1>
      </article>
    </a>
  )
}

export default styled(Card)` 
  background-color: ${({ theme }) => theme.primaryColorLight};
  transition: background ${({ theme }) => theme.transition};
  text-align: center;

  &:hover {
    background-color: ${({ theme }) => theme.primaryColorDark};
  }

  .card {
    padding: 10%;
  }

  .card__short-title {
    font-family: ${({ theme }) => theme.fontTitles}
    font-size: ${({ theme }) => theme.fontSizeXXL};
    font-weight: bold;
  }
`
