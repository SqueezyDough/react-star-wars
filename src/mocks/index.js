export const characterWithTitle = {
  title: 'title',
  url: 'http://swapi.dev/api/people/1/',
  props: {
    mass: 77,
    hair_color: 'blond',
    skin_color: 'fair',
    eye_color: 'blue',
    birth_year: '19bby',
    gender: 'male',
    homeworld: 'http://swapi.dev/api/planets/1/'
  }
}

export const characterWithName = {
  name: 'name',
  url: 'http://swapi.dev/api/people/1/',
  props: [{
    mass: 77,
    hair_color: 'blond',
    skin_color: 'fair',
    eye_color: 'blue',
    birth_year: '19bby',
    gender: 'male',
    homeworld: 'http://swapi.dev/api/planets/1/'
  }]
}

export const option = {
  value: {
    name: 'name',
    url: 'url'
  }
}

export const options = [option]

export const resource = {
  name: 'people'
}

export const loadingState = {
  start: true,
  loaders: 8
}
