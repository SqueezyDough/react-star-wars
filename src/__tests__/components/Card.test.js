import React from 'react'
import renderer from 'react-test-renderer'
import Card from '../../components/Card'
import { characterWithTitle, characterWithName } from '../../mocks/index'

it('renders correctly with title', () => {
  const tree = renderer
    .create(<Card character={characterWithTitle} />)
    .toJSON()

  expect(tree).toMatchSnapshot()
})

it('renders correctly with name', () => {
  const tree = renderer
    .create(<Card character={characterWithName} />)
    .toJSON()

  expect(tree).toMatchSnapshot()
})
