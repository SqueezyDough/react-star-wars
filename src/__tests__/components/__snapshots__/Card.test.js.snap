// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly with name 1`] = `
.c0 {
  -webkit-transition: background;
  transition: background;
  text-align: center;
}

.c0 .card {
  padding: 10%;
}

.c0 .card__short-title {
  font-weight: bold;
}

<a
  className="c0"
  href={"http://localhost/people/1/"}
>
  <article
    className="card"
  >
    <span
      className="card__short-title"
    >
      n
    </span>
    <h1
      className="card__title"
    >
      name
    </h1>
  </article>
</a>
`;

exports[`renders correctly with title 1`] = `
.c0 {
  -webkit-transition: background;
  transition: background;
  text-align: center;
}

.c0 .card {
  padding: 10%;
}

.c0 .card__short-title {
  font-weight: bold;
}

<a
  className="c0"
  href={"http://localhost/people/1/"}
>
  <article
    className="card"
  >
    <span
      className="card__short-title"
    >
      t
    </span>
    <h1
      className="card__title"
    >
      title
    </h1>
  </article>
</a>
`;
